### PR TITLE
Making margin parameterizable in nn.MultiMarginCriterion

### PR DESCRIPTION
**Pull request by nybbles**
This is https://github.com/torch/nn/pull/637, but rebased and with updates to the docs.

### CODE REVIEW

File: MultiMarginCriterion.lua [hunk 1]
```diff
@@ -1,9 +1,10 @@
 local THNN = require 'nn.THNN'
 local MultiMarginCriterion, parent = torch.class('nn.MultiMarginCriterion', 'nn.Criterion')
 
-function MultiMarginCriterion:__init(p, weights)
+function MultiMarginCriterion:__init(p, margin, weights)
```
Comment (soumith) [inline MultiMarginCriterion.lua]: please make margin to be the 3rd argument. the calls need to be backward-compatible.

Comment (nybbles) [inline MultiMarginCriterion.lua]: done, updated

**Comment by soumith** [comment]
Thanks nimalan.

**Comment by nybbles** [comment]
My pleasure!

**Pull request merged**